### DISCUSSION TITLE
Why is it not supported to modify the SpanContext in the Span, and what should I do if I need to resample？

### DISCUSSION THREAD

**Discussion by wuboui** [open-telemetry/opentelemetry-go#1760] [Q&A] [answered]

**Comment by Aneurysm9** [comment, marked as answer]
[The specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spancontext) requires that SpanContext is immutable.  Resampling a span in process is potentially dangerous since other spans may have been created from it assuming the original sampling decision which can result in broken traces or otherwise lost data.  It is generally only safe to either perform head sampling in the SDK or tail sampling, with all spans in a trace available for resampling, in the collector or other span aggregator.

**Reply by wuboui** [reply]
But if the downstream link hits the sampling, but the current rpc service is not hit, how to make the current service also be able to sample?

**Reply by wuboui** [reply]
How to sampling in tail?

**Reply by Aneurysm9** [reply]
To enable tail sampling you would need to always sample spans in the SDK and export them to a collector or other system configured to aggregate samples for a trace and perform tail sampling.  Tail sampling cannot be performed in-process with span creation.
